pkg/scalers/azure_monitor_scaler.go

```diff
@@ -191,7 +192,8 @@ func parseAzurePodIdentityParams(config *ScalerConfig) (clientID string, clientP
 		if len(clientPassword) == 0 {
 			return "", "", fmt.Errorf("no activeDirectoryClientPassword given")
 		}
-	} else if config.PodIdentity != kedav1alpha1.PodIdentityProviderAzure {
+	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
```

Comment:
Can we open an issue to support MI for Azure MOnitor?

Reply:
Monitor does support both the identities; I am not sure what you mean by this.

Reply:
This case is the same as default, which says to return an error or am I misinterpreting this?

Reply:
Unlike traditional C-like languages, Go switch statements do not fall through to the next case. So there's no need to explicitly `break`, this is the default behaviour in Go. To fall through to the next statement, you have to explicitly use `fallthrough`.

So, in this case, if they are using Pod / Workload Identity we don't need parse any params.